deepchecks/tabular/dataset.py

```diff
@@ -1061,6 +1066,158 @@ def datasets_share_date(cls, *datasets: 'Dataset') -> bool:
 
         return True
 
+    def _get_dataset_description(
+        self,
+        max_rows_to_show: int = 25,
+        max_columns_to_show: int = 7,
+        use_head_tail_separator: bool = False
+    ) -> t.Tuple[
+        pd.DataFrame,
+        pd.DataFrame
+    ]:
+        tail = 5 if max_rows_to_show > 15 else 2
+        head = max_rows_to_show - tail
+
+        rows_to_show = (
+            [*self.data.index[:head], *self.data.index[:-tail-1:-1]]
+            if len(self.data) > max_rows_to_show
+            else list(self.data.index)
+        )
+
+        data = self.data.iloc[rows_to_show, :].copy()
+
+        features = self.features
+        categorical_features = self.cat_features
+        numerical_features = self.numerical_features
+
+        label_column = t.cast(pd.Series, self.data[self.label_name]) if self.label_name else None
+        index_column = self.index_col
+        datetime_column = self.datetime_col
+
+        dataset_columns = []
+        dataset_columns_info = []
+
+        if index_column is not None:
+            index_name = index_column.name
+            data.loc[:, index_name] = index_column.loc[rows_to_show]
+            index_column_info = [index_name, infer_dtype(index_column, skipna=True), 'Index']
+            dataset_columns.append(index_name)
+            dataset_columns_info.append(index_column_info)
+        else:
+            index_name = None
+
+        if datetime_column is not None:
+            datetime_name = datetime_column.name
+            data.loc[:, datetime_name] = datetime_column.loc[rows_to_show]
+            datetime_column_info = [datetime_name, infer_dtype(datetime_column, skipna=True), 'Datetime']
+            dataset_columns.append(datetime_name)
+            dataset_columns_info.append(datetime_column_info)
+        else:
+            datetime_name = None
+
+        label_name = label_column.name if label_column is not None else None
+
+        for feature_name in t.cast(t.List[str], features):
+            if feature_name in (index_name, datetime_name, label_name):
```

Comment:
I think it shouldn't be possible for a feature to be an index, date or label (we're supposed to check for it in Dataset construction)

Reply:
the constructor will raise an error in this case

```python
        if self._label_name in self.features:
            raise DeepchecksValueError(f'label column {self._label_name} can not be a feature column')

        if self._datetime_name in self.features:
            raise DeepchecksValueError(f'datetime column {self._datetime_name} can not be a feature column')

        if self._index_name in self.features:
            raise DeepchecksValueError(f'index column {self._index_name} can not be a feature column')
```